tests/geobase/tests_FairGeoSet.cxx

```diff
@@ -0,0 +1,103 @@
+/********************************************************************************
+ *   Copyright (C) 2024 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH     *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+#include "FairGeoBrik.h"
+#include "FairGeoMedia.h"
+#include "FairGeoMedium.h"
+#include "FairGeoShapes.h"
+#include "FairTestGeo.h"
+#include "checks_FairGeoSet.h"
+
+#include <TList.h>
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators_all.hpp>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#ifndef TESTS_GEOBASE_FAIRGEOSET_H
+#define TESTS_GEOBASE_FAIRGEOSET_H
+
+TEST_CASE("FairTestGeo")
+{
+    /*
+        std::ostringstream os;
+
+        os << "//----------------------------------------------------------\n";
+        os << "air                3  14.01  16.  39.95  7.  8.  18.  1.205e-3  .755  .231.014\n";
+        os << "                   0  1  30.  .001\n";
+        os << "silicon            1  28.0855 14.0 2.33\n";
+        os << "                   1  1  20.  .001\n";
+        os << "                   0";
+        std::istringstream mediafile( os.str() );
+    */
+
+    std::fstream mediafile{};
+    mediafile.open("data/testmedia.geo", std::ios::in);
+
+    std::fstream geofile{};
+    geofile.open("data/testdet_brick.geo", std::ios::in);
+
+    if (!geofile.is_open()) {
+        INFO("Geo file isn't open");
+        abort();
+    }
+    if (!mediafile.is_open()) {
+        INFO("Media file isn't open");
+        abort();
+    }
+
+    REQUIRE(geofile.is_open());
+    REQUIRE(mediafile.is_open());
+
+    INFO("Media and geo files open");
+
+    FairGeoMedia media;
+    media.read(mediafile);
+
+    INFO("Media file read");
+
+    FairGeoNode* volu = new FairGeoNode();
+    volu->SetName("cave");
+    volu->setVolumeType(kFairGeoTopNode);
+    volu->setActive();
+    volu->setShape(new FairGeoBrik());
+    FairGeoMedium* medium = media.getMedium("air");
+    if (!medium) {
+        medium = new FairGeoMedium();
+        media.addMedium(medium);
+    }
+    volu->setMedium(medium);
```

Comment:
```suggestion
    if (auto air = media.getMedium("air"); air) {
        volu->setMedium(air);
    } else {
        auto generic = std::make_unique<FairGeoMedium>();
        volu->setMedium(generic.get());
        media.addMedium(generic.release());
    }
```

Let's not re-use the same raw pointer for non-owned and owned objects.

Reply:
Could you explain the following construct

`if (auto air = media.getMedium("air"); air) `

Reply:
https://en.cppreference.com/w/cpp/language/if - scroll down to the box *If statements with initializer* (sry, it won't let me link to this section directly). It is a C++17 feature, basically adds a init section to `if` like in the `for` statement. This way one does not need to declare variables in the outer scope (to the if statement), if this variable is only used within the if statement scope.

Reply:
Great. I was not aware of this feature.

Reply:
Done